web/src/lib/components/photos-page/delete-asset-dialog.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+
+  import ConfirmDialogue from '../shared-components/confirm-dialogue.svelte';
+  import { showDeleteModal } from '$lib/stores/preferences.store';
+
+  export let size: number;
+
+  let checked = false;
+
+  const onToggle = () => {
+    checked = !checked;
+  };
+
+  const handleConfirm = () => {
+    if (checked) {
+      $showDeleteModal = false;
+    }
+    dispatch('confirm');
+  };
+
+  const dispatch = createEventDispatcher<{
+    confirm: void;
+    cancel: void;
+    escape: void;
+  }>();
+</script>
+
+<ConfirmDialogue
+  title="Permanently Delete Asset{size > 1 ? 's' : ''}"
+  confirmText="Delete"
+  on:confirm={handleConfirm}
+  on:cancel={() => dispatch('cancel')}
+  on:escape={() => dispatch('escape')}
```

Comment:
I don't think we need separate cancel vs escape events, just emit "close" or "cancel" both both of them.